pkg/engine/executor/limit_test.go

```diff
@@ -4,11 +4,199 @@ import (
+		pipeline := c.executeLimit(ctx, &physical.Limit{}, nil)
+		err := pipeline.Read()
+		require.Equal(t, EOF, err)
+	})
+
+	t.Run("with multiple inputs", func(t *testing.T) {
+		pipeline := c.executeLimit(ctx, &physical.Limit{}, []Pipeline{emptyPipeline(), emptyPipeline()})
+		err := pipeline.Read()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "limit expects exactly one input, got 2")
+	})
+
+	t.Run("with valid input", func(t *testing.T) {
+		// Create test data
+		fields := []arrow.Field{
+			{Name: "name", Type: arrow.BinaryTypes.String},
+		}
+		csvData := "Alice\nBob\nCharlie"
+		record, err := CSVToArrow(fields, csvData)
+		require.NoError(t, err)
+		defer record.Release()
+
+		record.Retain()
+		source := NewBufferedPipeline(record)
+
+		// Execute limit with skip=1 and fetch=1
+		limit := &physical.Limit{Skip: 1, Fetch: 1}
+		pipeline := c.executeLimit(ctx, limit, []Pipeline{source})
+		defer pipeline.Close()
+
+		// Read from the pipeline
+		err = pipeline.Read()
+		require.NoError(t, err)
+
+		batch, err := pipeline.Value()
+		require.NoError(t, err)
+		require.NotNil(t, batch)
+
+		// Should have returned only the second row
+		require.Equal(t, int64(1), batch.NumRows())
+		require.Equal(t, int64(1), batch.NumCols())
+		require.Equal(t, "name", batch.ColumnName(0))
+		require.Equal(t, "Bob", batch.Column(0).ValueStr(0))
+
+		// Next read should return EOF
+		err = pipeline.Read()
+		require.Equal(t, EOF, err)
+	})
+
+	// Test with desired rows split across 2 batches
+	t.Run("with rows split across batches", func(t *testing.T) {
+		// Create schema with a single string column
+		fields := []arrow.Field{
+			{Name: "letter", Type: arrow.BinaryTypes.String},
+		}
+
+		// Create first batch with letters A-C
+		batch1Data := "A\nB\nC"
+		batch1, err := CSVToArrow(fields, batch1Data)
+		require.NoError(t, err)
+		defer batch1.Release()
+
+		// Create second batch with letters D-F
+		batch2Data := "D\nE\nF"
+		batch2, err := CSVToArrow(fields, batch2Data)
+		require.NoError(t, err)
+		defer batch2.Release()
+
+		// Create source pipeline with both batches
+		batch1.Retain()
+		batch2.Retain()
+		source := NewBufferedPipeline(batch1, batch2)
+
+		// Create limit pipeline that skips 2 and fetches 3
+		// Should return C from batch1, and D,E from batch2
+		limit := &physical.Limit{Skip: 2, Fetch: 3}
+		pipeline := c.executeLimit(ctx, limit, []Pipeline{source})
+		defer pipeline.Close()
+
+		// First read should give us C from batch1
+		expectedFields := []arrow.Field{
+			{Name: "letter", Type: arrow.BinaryTypes.String},
+		}
+
+		expectedData := "C\nD\nE"
+		expectedRecord, err := CSVToArrow(expectedFields, expectedData)
+		require.NoError(t, err)
+		defer expectedRecord.Release()
+
+		expectedPipeline := NewBufferedPipeline(expectedRecord)
+		defer expectedPipeline.Close()
+
+		AssertPipelinesEqual(t, pipeline, expectedPipeline)
+	})
+}
+
+func TestLimitPipeline_Skip_Fetch(t *testing.T) {
```

Comment:
nit: the old `TestLimit` might not be needed anymore?

Reply:
You're right, but I'll leave that for later